### PR TITLE
geo_shape provider: fixed calculation

### PR DESCRIPTION
**Pull request by chaudum**
of destination points based on distance and bearing
and do not allow areas that cover north or south pole

```
dba:cr8 (h/geoshape*) $ env/bin/cr8 insert-fake-data --hosts st1.p.fir.io:4200 --table geo --num-records 1000000
Found schema: 
{
    "area": "geo_shape",
    "id": "integer"
}
Using insert statement: 
insert into "doc"."geo" ("area", "id") values (?, ?)
Will make 1000 requests with a bulk size of 1000
Generating fake data and executing inserts
100%|███████████████████████████████████████████████████████████████████| 1000/1000 [13:47<00:00,  1.52s/ requests]
```

fixes https://github.com/mfussenegger/cr8/issues/185

### CODE REVIEW

File: cr8/fake_providers.py [hunk 1]
```diff
@@ -83,31 +113,25 @@ def geo_shape(self, sides=5, center=None, distance=None):
...
         else:
             # 6371 => earth radius in km
             # assert that shape radius is maximum half of earth's circumference
             assert isinstance(distance, int)
-            assert distance <= self.EARTH_RADIUS * math.pi
+            assert distance <= EARTH_RADIUS * math.pi
 
-        d_arc = distance * 180.0 / self.EARTH_RADIUS / math.pi
+        if center is None:
+            # required minimal sherical distance from north/southpole
```
Comment (mfussenegger) [inline cr8/fake_providers.py]: typo in `sherical`?

File: cr8/fake_providers.py [hunk 2]
```diff
@@ -83,31 +113,25 @@ def geo_shape(self, sides=5, center=None, distance=None):
...
+            dp = distance * 180.0 / EARTH_RADIUS / math.pi
+            center = self.geo_point(lat_min=-90.0 + dp, lat_max=90.0 - dp)
+        else:
+            assert -180.0 <= center[0] <= 180.0
+            assert -90.0 <= center[1] <= 90.0
```
Comment (mfussenegger) [inline cr8/fake_providers.py]: maybe add assertion messages?

File: cr8/fake_providers.py [hunk 3]
```diff
@@ -28,6 +29,36 @@ def auto_inc(fake):
     return Counter(manager.Value('i', 0), manager.Lock())
 
 
+def _dest_point(point, distance, bearing, radius):
+    # calculation taken from
+    # https://cdn.rawgit.com/chrisveness/geodesy/v1.1.2/latlon-spherical.js
+    # https://www.movable-type.co.uk/scripts/latlong.html
+
+    δ = distance / radius  # angular distance in rad
```
Comment (mfussenegger) [inline cr8/fake_providers.py]: :scream:

Comment (chaudum) [reply cr8/fake_providers.py]: would you rather use descriptive variable names?

Comment (mfussenegger) [reply cr8/fake_providers.py]: No I think it's okay since the formulas use these symbols as well =)

**Pull request merged**